reference/extensions/hooks.rst

```diff
@@ -1,4 +1,172 @@
+   :caption: *hook_full.py*
+
+    def pre_export(conanfile):
+        conanfile.output.info("Running before to execute export() method.")
+
+    def post_export(conanfile):
+        conanfile.output.info("Running after of executing export() method.")
+
+    def pre_source(conanfile):
+        conanfile.output.info("Running before to execute source() method.")
+
+    def post_source(conanfile):
+        conanfile.output.info("Running after of executing source() method.")
+
+    def pre_generate(conanfile):
+        conanfile.output.info("Running before to execute generate() method.")
+
+    def post_generate(conanfile):
+        conanfile.output.info("Running after of executing generate() method.")
+
+    def pre_build(conanfile):
+        conanfile.output.info("Running before to execute build() method.")
+
+    def post_build(conanfile):
+        conanfile.output.info("Running after of executing build() method.")
+
+    def pre_package(conanfile):
+        conanfile.output.info("Running before to execute package() method.")
+
+    def post_package(conanfile):
+        conanfile.output.info("Running after of executing package() method.")
+
+    def pre_package_info(conanfile):
+        conanfile.output.info("Running before to execute package_info() method.")
+
+    def post_package_info(conanfile):
+        conanfile.output.info("Running after of executing package_info() method.")
+
+Functions of the hooks are intended to be self-descriptive regarding to the execution of them. For example, the ``pre_package()`` function
+is called just before the ``package()`` method of the recipe is executed.
+
+
+Function parameters
+-------------------
+
+All hook methods are filled only with the same single object:
+
+- **conanfile**: It is a regular ``ConanFile`` object loaded from the recipe that received the Conan command. It has its normal attributes
+  and dynamic objects such as ``build_folder``, ``package_folder``, ``output``, ``dependencies``,  ``options`` ...
+
+Storage, activation and sharing
+-------------------------------
+
+Hooks are Python files stored under *~/.conan2/extensions/hooks* folder and **their file name should start with hook_ and end with the
+.py extension**.
+
+The activation of the hooks is done automatically once the hook file is stored in the hook folder.
+In case storing in subfolders, it works automatically too.
+
+To deactivate a hook, its file should be removed from the hook folder. There is no configuration which can deactivate but keep the file stored in hooks folder.
+
+Hooks are considered part of the Conan client configuration and can be shared as usual with the `reference_commands_config` command.
+However, they can also be managed in isolated Git repositories cloned into the *~/.conan2/extensions/hooks* folder:
+
+.. code-block:: bash
+
+    $ cd ~/.conan2/extensions/hooks
+    $ git clone https://github.com/myuser/my-conan-hooks.git my_hooks
+
+This way you can easily change from one version to another.
```

Comment:
Are we sure about this, and like this is a good practice? I wouldn't say so, better to remove it until we discuss it.
Because it can open the door to having everything, profiles, settings.yml, conf, etc also as git repositories cloned directly into the Conan cache.

Reply:
Removed.